### DISCUSSION TITLE
No API/commands to toggle between collapsed/visable?

### DISCUSSION THREAD

**Discussion by worksofliam** [microsoft/vscode-discussions#1023] [Extension Development QnA] [answered]
So it seems like there is still no command to collapse a specific view. Let's say I have a view like this:

```
    "views": {
      "ibmi-panel": [
        {
          "type": "webview",
          "id": "vscode-db2i.resultset",
          "name": "Results",
          "when": "code-for-ibmi:connected == true",
          "contextualTitle": "IBM i"
        },
```

I can use a custom context to hide and show a view, but I don't want to do that here because I want the webview to not be disposed. The user can manually collapse which doesn't dispose of the webview, but there's no command to do that? There is `vscode-db2i.resultset.toggleVisibility`, but that's fully hiding it and can't be expanded. What am I missing? Why is this missing? I even checked the VS Code codebase and it looks like it's not doable.

I really want to programmatically control the visibility property here between `visible` and `collapsed`:

![image](https://github.com/microsoft/vscode-discussions/assets/3708366/af1e59b7-0118-4320-ae2c-961d2e0f633b)

**Comment by isidorn** [comment]
Not sure if possible. @alexr00 might be able to help here.

**Comment by alexr00** [comment, marked as answer]
Feature request is here: https://github.com/microsoft/vscode/issues/88219
